### ISSUE TITLE
log spam + memory leak since v2.2.2

### ISSUE THREAD

**Issue by superxor** [rancher/rancher#19831]
Ubuntu 16.04 + 17.03.3~ce-0~ubuntu-xenial
Rancher v2.2.2

Since the upgrade we're experiencing massive log spam looking like this:

> 2019/04/26 10:37:14 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.24:52494: i/o timeout
2019/04/26 10:37:14 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.96:50488: i/o timeout
2019/04/26 10:37:15 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.133:34962: i/o timeout
2019/04/26 10:37:15 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.104:47320: i/o timeout
2019/04/26 10:37:15 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.105:36276: i/o timeout
2019/04/26 10:37:16 [INFO] Handling backend connection request [c-q89w5:m-76075bccca21]
2019/04/26 10:37:16 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.23:44106: i/o timeout
2019/04/26 10:37:17 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.137:50636: i/o timeout
2019/04/26 10:37:17 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.108:57132: i/o timeout
2019/04/26 10:37:17 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.139:60740: i/o timeout
2019/04/26 10:37:17 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.12:51888: i/o timeout
2019/04/26 10:37:17 [INFO] Handling backend connection request [c-q89w5:m-bfdc2014d1b5]
2019/04/26 10:37:17 [INFO] error in remotedialer server [400]: read tcp 172.17.0.2:443->10.3.1.138:57304: i/o timeout

Furthermore the memory is going up from 1GB to 8GB every 2 minutes, falling back at once.
Responsible being: tiller --listen :51754 --probe :48635

webgui is extremely laggy and mostly unresponsive.

I thought v2.2.2 is supposed to be stable?

**Comment by alena1108** [comment]
@xorpierre Did you just upgrade from the previous version of Rancher and started seeing the problem? Or did you turn on any of Rancher 2.2.2 new features, like monitoring or global dns?

**Comment by xorpierre** [comment]
You're right...colleague enabled the new cluster monitoring feature which seems to be the cause for this. From what I read so far this is in early alpha stage.
Guess cause is known then

**Comment by alena1108** [comment]
@xorpierre I'd like to reopen the issue. Cluster monitoring is a GA feature on 2.2.2, and we should investigate on what's going on.

@loganhz looks like the way we start tiller is not ideal:

https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common.go#L61

I wonder if when the connection to a user cluster is unstable, the io/timeout can cause the command to be retried over and over again w/o actually killing the last one: https://github.com/rancher/rancher/blob/c37a59d761245b483d4f8775794846f2a1effae8/pkg/controllers/user/helm/common/common.go#L89

**Comment by daanemanz** [comment]
We're seeing very similar messages on our Rancher servers (also v2.2.2) but we _don't_ have monitoring enabled. We disabled it, but the prometheus and alertmanager pods still seem to be running.

**Comment by deniseschannon** [comment]
@xorpierre @daanemanz could you test this out with v2.2.3-rc6 (do not use v2.2.3-rc7)? We have made some changes regarding tiller backoff so that it wouldn't loop continuously.

**Comment by daanemanz** [comment]
@deniseschannon I'll test it out with v2.2.3-rc6 but can I downgrade back to v2.2.2 easily? RC versions are unsupported...

**Comment by deniseschannon** [comment]
@daanemanz In theory, downgrading would be easy, but it hasn't been tested. I was hoping you would be able to spin up something new as we haven't been able to easily reproduce.

**Comment by orangedeng** [comment]
@alena1108 Since we don't have a stable way to reproduce this issue, I don't think it is necessary to print error logs which tiller returns.

**Comment by xorpierre** [comment]
was this seen by the team? related https://github.com/rancher/rancher/issues/17617

**Comment by alena1108** [comment]
@orangedeng 
> Since we don't have a stable way to reproduce this issue, I don't think it is necessary to print error logs which tiller returns.

But do you think it would make it easier to investigate why the failure happens? If lets say the user hits it, and provided us with the log.

**Comment by orangedeng** [comment]
As we put tiller stdout and stderr to Rancher console,
https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common/common.go#L65-L79
We can only get the killing process err return in https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common.go#L61

**Comment by cjellick** [comment]
i merged the pr for printing the error, but i havent moved this issue over bc i dont know what the resolution is supposed to be

**Comment by alena1108** [comment]
@cjellick the expectations are that the issue was mitigated in 2.2.4 by:

* Reducing the number of unnecessary helm deploy retries
* Fixing the bug when monitoring was constantly redeployed even when nothing got changed

As we couldn't reliably reproduce the issue, I'll keep it open, but move it out of the milestone. We'd need @superxor help in validating on v2.2.4 once it ships.

**Comment by superxor** [comment]
I'm happy to upgrade if the release is deemed stable, I just cant run patched test versions on our production environment. But what I read so far seemed logically and explained especially the memory leak which was the biggest issue.

But we're dealing with some other issues right now which seem way more important that some new fancy feature, because base functions are not fully working. See my other opened issue, especially the one about being unable to add new worker nodes, that's a complete show-stopper

**Comment by deniseschannon** [comment]
@superxor can you link what your other issue is?

**Comment by superxor** [comment]
@deniseschannon https://github.com/rancher/rancher/issues/20232 and I am not alone with that issue, several related comments in the issue opened by someone else. This is a new issue, not necessarily connected to v.2.2.2 it seems but its a big issue.

**Comment by Remigius2011** [comment]
Hi,

My rancher v2.2.2 single node install got unresponsive several times recently. No I dropped onto this issue because I have observed a tiller process on the node that hosts rancher, which intermittedly consumes practically all available memory, then goes back to a low percentage.

As low memory was the only cause of instability of either rancher itself or cluster nodes that I have opbserved so far, I think there is a strong interest in keeping memory consumption of all components as low as possible. I don't exactly know how tiller is used on the context of rancher, but consuming up to three times as much memory as rancher itself for keeping up (in my case two) helm catalogs is imho somewhat overshooting.

Can - as a work-around - the helm catalogs be deactivated temporarily in order to lower memory consumption without losing the ability to administer deployed help apps?
